app/uk/gov/hmrc/thirdpartyapplication/models/ApplicationSearch.scala

```diff
@@ -30,7 +30,9 @@ case class ApplicationSearch(pageNumber: Int = 1,
                              textToSearch: Option[String] = None,
                              apiContext: Option[ApiContext] = None,
                              apiVersion: Option[ApiVersion] = None,
-                             sort: ApplicationSort = SubmittedAscending)
+                             sort: ApplicationSort = SubmittedAscending){
+                               def hasSubscriptionFilter() = filters.exists(filter => filter.isInstanceOf[APISubscriptionFilter])
```

Comment:
We don't tend to us isInstanceOf or asInstanceOf as they are not great.

Maybe try this instead?

`case class ApplicationSearch(
  pageNumber: Int = 1,
  pageSize: Int = Int.MaxValue,
  filters: List[ApplicationSearchFilter] = List.empty,
  textToSearch: Option[String] = None,
  apiContext: Option[ApiContext] = None,
  apiVersion: Option[ApiVersion] = None,
  sort: ApplicationSort = SubmittedAscending
){
  def hasSubscriptionFilter() = filters.exists {
    case _: APISubscriptionFilter => true
    case _ => false
  }
}
`

Reply:
Formatting is rubbish